### PR TITLE
change storage blob track1 to track2

### PR DESCRIPTION
**Pull request by Luyunmt**
Change storage blob package track1 to track2.

### CODE REVIEW

**Comment by v-xuto** [comment]
@jongio - Can you help ping the corresponding person to review this PR?

**Comment by v-xuto** [comment]
@scottaddie, @AlexGhiondea, @kyle-patterson , @tg-msft, @amishra-dev - Could you help review this PR?

File: SasTutorial/Program.cs [hunk 1]
```diff
@@ -167,105 +170,132 @@ static void Main()
         /// Returns a URI containing a SAS for the blob container.
         /// </summary>
         /// <param name="container">A reference to the container.</param>
-        /// <param name="storedPolicyName">A string containing the name of the stored access policy. If null, an ad-hoc SAS is created.</param>
+        /// <param name="StorageSharedKeyCredential">Storage Shared Key Credential.</param>
         /// <returns>A string containing the URI for the container, with the SAS token appended.</returns>
-        static string GetContainerSasUri(CloudBlobContainer container, string storedPolicyName = null)
+        static UriBuilder GetContainerSasUri(BlobContainerClient container, StorageSharedKeyCredential storageSharedKeyCredential)
```
Comment (tg-msft) [inline SasTutorial/Program.cs:173]: We should remove this method and use `container.GenerateSasUri(...)` instead.  The access policy example can show how to use `BlobSasBuilder` for more advanced scenarios.

File: SasTutorial/Program.cs [hunk 2]
```diff
@@ -18,8 +18,12 @@
 using System.Text;
 using System.IO;
 using Microsoft.Azure;
```
Comment (tg-msft) [inline SasTutorial/Program.cs]: Why is this still here?  We've probably got an extra dependency hanging around too if the compiler didn't complain.

File: SasTutorial/Program.cs [hunk 3]
```diff
@@ -18,8 +18,12 @@
...
-using Microsoft.Azure.Storage;
-using Microsoft.Azure.Storage.Blob;
+using Azure.Storage.Blobs;
+using Azure.Storage;
+using Azure.Storage.Sas;
+using System.Net;
+using Azure.Storage.Blobs.Models;
+using System.Collections.Generic;
```
Comment (tg-msft) [inline SasTutorial/Program.cs]: Sort the namespaces

File: SasTutorial/Program.cs [hunk 4]
```diff
@@ -77,34 +81,31 @@ static void Main()
...
             //Get a reference to a container to use for the sample code, and create it if it does not exist.
-            CloudBlobContainer container = blobClient.GetContainerReference(containerName);
+            BlobContainerClient container = blobServiceClient.GetBlobContainerClient(containerName);
 
             try 
             { 
                 container.CreateIfNotExists();
             }
-            catch (StorageException)
+            catch (Exception)
```
Comment (tg-msft) [inline SasTutorial/Program.cs]: Can we change this to `RequestFailedException` to let people know what to expect?

File: SasTutorial/Program.cs [hunk 5]
```diff
@@ -167,105 +170,132 @@ static void Main()
...
         /// <summary>
         /// Returns a URI containing a SAS for the blob.
         /// </summary>
         /// <param name="container">A reference to the container.</param>
         /// <param name="blobName">A string containing the name of the blob.</param>
-        /// <param name="policyName">A string containing the name of the stored access policy. If null, an ad-hoc SAS is created.</param>
+        /// <param name="StorageSharedKeyCredential">Storage Shared Key Credential.</param>
         /// <returns>A string containing the URI for the blob, with the SAS token appended.</returns>
-        static string GetBlobSasUri(CloudBlobContainer container, string blobName, string policyName = null)
+        static UriBuilder GetBlobSasUri(BlobContainerClient container, string blobName, StorageSharedKeyCredential storageSharedKeyCredential)
```
Comment (tg-msft) [inline SasTutorial/Program.cs]: Similarly we should replace this method with `blob.GenerateSasUri()` but leave the access policy one to show advanced use cases.

File: SasTutorial/Program.cs [hunk 6]
```diff
@@ -293,12 +323,12 @@ static void TestContainerSAS(string sasUri, string blobName, string blobContent)
                 msWrite.Position = 0;
                 using (msWrite)
                 {
-                    blob.UploadFromStream(msWrite);
+                    blob.Upload(msWrite);
```
Comment (tg-msft) [inline SasTutorial/Program.cs]: Let's replace all this stream stuff with `blob.Upload(BinaryData.FromString(blobContent))`.

File: SasTutorial/Program.cs [hunk 7]
```diff
@@ -293,12 +323,12 @@ static void TestContainerSAS(string sasUri, string blobName, string blobContent)
...
                 }
                 Console.WriteLine("Write operation succeeded for SAS " + sasUri);
                 Console.WriteLine();
             }
-            catch (StorageException e)
+            catch (Exception e)
```
Comment (tg-msft) [inline SasTutorial/Program.cs]: catch `RequestFailedException` here and everywhere else

File: SasTutorial/Program.cs [hunk 8]
```diff
@@ -414,26 +436,21 @@ static void TestBlobSAS(string sasUri, string blobContent)
...
-                        string line;
-                        while ((line = reader.ReadLine()) != null)
-                        {
-                            Console.WriteLine(line);
-                        }
+                        Console.WriteLine(line);
                     }
-                    Console.WriteLine();
                 }
+                Console.WriteLine();
```
Comment (tg-msft) [inline SasTutorial/Program.cs]: Let's replace this whole thing with `blob.DownloadContent()` which gives back `BinaryData` that has a `GetString` method we can call to get the entire thing as text.  Then we can just `Console.WriteLine` the entire payload at once.

File: SasTutorial/SasTutorial.csproj [hunk 9]
```diff
@@ -33,14 +33,17 @@
...
-    <Reference Include="Microsoft.Azure.Storage.Common, Version=10.0.2.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\packages\Microsoft.Azure.Storage.Common.10.0.2\lib\net452\Microsoft.Azure.Storage.Common.dll</HintPath>
+    <Reference Include="Azure.Storage.Common, Version=12.8.0.0, Culture=neutral, PublicKeyToken=92742159e12e44c8, processorArchitecture=MSIL">
+      <HintPath>..\packages\Azure.Storage.Common.12.8.0\lib\netstandard2.0\Azure.Storage.Common.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.Bcl.AsyncInterfaces, Version=1.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.Bcl.AsyncInterfaces.1.0.0\lib\net461\Microsoft.Bcl.AsyncInterfaces.dll</HintPath>
     </Reference>
     <Reference Include="Microsoft.Data.Edm, Version=5.8.4.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <HintPath>..\packages\Microsoft.Data.Edm.5.8.4\lib\net40\Microsoft.Data.Edm.dll</HintPath>
```
Comment (tg-msft) [inline SasTutorial/SasTutorial.csproj]: Why is this here?  It's not needed by Track 2 Storage.

File: SasTutorial/SasTutorial.csproj [hunk 10]
```diff
@@ -54,14 +57,42 @@
     <Reference Include="Microsoft.WindowsAzure.Configuration, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <HintPath>..\packages\Microsoft.WindowsAzure.ConfigurationManager.3.2.3\lib\net40\Microsoft.WindowsAzure.Configuration.dll</HintPath>
     </Reference>
```
Comment (tg-msft) [inline SasTutorial/SasTutorial.csproj]: Remove this too

File: SasTutorial/SasTutorial.csproj [hunk 11]
```diff
@@ -54,14 +57,42 @@
...
+    <Reference Include="System.Numerics" />
+    <Reference Include="System.Numerics.Vectors, Version=4.1.4.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>..\packages\System.Numerics.Vectors.4.5.0\lib\net46\System.Numerics.Vectors.dll</HintPath>
+    </Reference>
+    <Reference Include="System.Runtime.CompilerServices.Unsafe, Version=4.0.5.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>..\packages\System.Runtime.CompilerServices.Unsafe.4.6.0\lib\netstandard2.0\System.Runtime.CompilerServices.Unsafe.dll</HintPath>
+    </Reference>
     <Reference Include="System.Spatial, Version=5.8.4.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <HintPath>..\packages\System.Spatial.5.8.4\lib\net40\System.Spatial.dll</HintPath>
     </Reference>
```
Comment (tg-msft) [inline SasTutorial/SasTutorial.csproj]: We don't need this

File: SasTutorial/SasTutorial.csproj [hunk 12]
```diff
@@ -54,14 +57,42 @@
...
+      <HintPath>..\packages\System.Text.Json.4.6.0\lib\net461\System.Text.Json.dll</HintPath>
+    </Reference>
+    <Reference Include="System.Threading.Tasks.Extensions, Version=4.2.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
+      <HintPath>..\packages\System.Threading.Tasks.Extensions.4.5.2\lib\netstandard2.0\System.Threading.Tasks.Extensions.dll</HintPath>
+    </Reference>
+    <Reference Include="System.ValueTuple, Version=4.0.3.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
+      <HintPath>..\packages\System.ValueTuple.4.5.0\lib\net47\System.ValueTuple.dll</HintPath>
+    </Reference>
     <Reference Include="System.Xml.Linq" />
     <Reference Include="System.Data.DataSetExtensions" />
```
Comment (tg-msft) [inline SasTutorial/SasTutorial.csproj]: I'm not sure either of these are needed?

File: SasTutorial/packages.config [hunk 13]
```diff
@@ -1,12 +1,22 @@
...
-  <package id="Microsoft.Azure.Storage.Blob" version="10.0.2" targetFramework="net472" />
-  <package id="Microsoft.Azure.Storage.Common" version="10.0.2" targetFramework="net472" />
+  <package id="Azure.Core" version="1.18.0" targetFramework="net472" />
+  <package id="Azure.Storage.Blobs" version="12.9.1" targetFramework="net472" />
+  <package id="Azure.Storage.Common" version="12.8.0" targetFramework="net472" />
+  <package id="Microsoft.Bcl.AsyncInterfaces" version="1.0.0" targetFramework="net472" />
   <package id="Microsoft.Data.Edm" version="5.8.4" targetFramework="net45" />
   <package id="Microsoft.Data.OData" version="5.8.4" targetFramework="net45" />
   <package id="Microsoft.Data.Services.Client" version="5.8.4" targetFramework="net45" />
   <package id="Microsoft.WindowsAzure.ConfigurationManager" version="3.2.3" targetFramework="net45" />
```
Comment (tg-msft) [inline SasTutorial/packages.config]: Remove all these

File: SasTutorial/packages.config [hunk 14]
```diff
@@ -1,12 +1,22 @@
...
-  <package id="Newtonsoft.Json" version="12.0.2" targetFramework="net45" />
+  <package id="System.Buffers" version="4.5.1" targetFramework="net472" />
+  <package id="System.Diagnostics.DiagnosticSource" version="4.6.0" targetFramework="net472" />
+  <package id="System.Memory" version="4.5.4" targetFramework="net472" />
+  <package id="System.Memory.Data" version="1.0.2" targetFramework="net472" />
+  <package id="System.Numerics.Vectors" version="4.5.0" targetFramework="net472" />
+  <package id="System.Runtime.CompilerServices.Unsafe" version="4.6.0" targetFramework="net472" />
   <package id="System.Spatial" version="5.8.4" targetFramework="net45" />
```
Comment (tg-msft) [inline SasTutorial/packages.config]: Remove this

File: SasTutorial/Program.cs (hunk 3 above)
Comment (Tianyun-G) [reply SasTutorial/Program.cs]: Updated.

File: SasTutorial/Program.cs (hunk 4 above)
Comment (Tianyun-G) [reply SasTutorial/Program.cs]: Updated.

File: SasTutorial/Program.cs (hunk 6 above)
Comment (Tianyun-G) [reply SasTutorial/Program.cs]: Updated.

File: SasTutorial/Program.cs (hunk 7 above)
Comment (Tianyun-G) [reply SasTutorial/Program.cs]: Updated.

File: SasTutorial/SasTutorial.csproj (hunk 9 above)
Comment (Tianyun-G) [reply SasTutorial/SasTutorial.csproj]: Updated.

File: SasTutorial/SasTutorial.csproj (hunk 11 above)
Comment (Tianyun-G) [reply SasTutorial/SasTutorial.csproj]: Updated.

File: SasTutorial/SasTutorial.csproj (hunk 12 above)
Comment (Tianyun-G) [reply SasTutorial/SasTutorial.csproj]: Updated.

File: SasTutorial/Program.cs (hunk 8 above)
Comment (Tianyun-G) [reply SasTutorial/Program.cs]: Updated.

File: SasTutorial/Program.cs (hunk 2 above)
Comment (Tianyun-G) [reply SasTutorial/Program.cs]: Updated.

File: SasTutorial/SasTutorial.csproj (hunk 10 above)
Comment (Tianyun-G) [reply SasTutorial/SasTutorial.csproj]: Updated.

File: SasTutorial/packages.config (hunk 13 above)
Comment (Tianyun-G) [reply SasTutorial/packages.config]: Updated.

File: SasTutorial/packages.config (hunk 14 above)
Comment (Tianyun-G) [reply SasTutorial/packages.config]: Updated.

File: SasTutorial/Program.cs (hunk 1 above)
Comment (v-yilinhu) [reply SasTutorial/Program.cs:173]: @tg-msft Updated.

File: SasTutorial/Program.cs (hunk 5 above)
Comment (v-yilinhu) [reply SasTutorial/Program.cs]: @tg-msft Updated.

File: SasTutorial/Program.cs [hunk 15]
```diff
@@ -115,46 +114,48 @@ static void Main()
...
+            UriBuilder storeContainerSAS = GetContainerSasUri(container, storeAccessPolicyName, storageSharedKeyCredential);
+            Console.WriteLine("1. SAS for blob container : " + storeContainerSAS);
             Console.WriteLine();
 
             //Test the SAS to ensure it works as expected.
-            //The write and list operations should succeed, and the read and delete operations should fail.
-            TestContainerSAS(adHocContainerSAS, blobName1, blobContent1);
+            //The write, read, and delete operations should  succeed, and the list operations should fail.
+            TestContainerSAS(storeContainerSAS.Uri, blobName1, blobContent1);
             Console.WriteLine();
```
Comment (tg-msft) [inline SasTutorial/Program.cs]: You should move this whole section to be 2.  The first demo should be the simple `container.GenerateSasUri(...)` like you have below, then show off the more advanced access policy second.  Ditto for blobs below.  Show a simple blob SAS first, then do the more advanced access policy blob SAS second.

**Comment by v-xuto** [comment]
@tg-msft  We have updated, please re-review this PR.

**Review by tg-msft — APPROVED** [review]
Left a comment that'd be great to address, but approved the change overall.

**Comment by v-xuto** [comment]
@tg-msft We have updated, please re-review this PR. If there is no problem, please help merge this PR.
